src/common/utxobased/db/newProcessor.ts

```diff
@@ -161,26 +194,160 @@ export async function makeNewProcessor(
+        }
+
+        transaction.ourAmount = calculateTxAmount(transaction)
+
+        // Save tx by blockheight
+        const txidsByTransactionBlockHeight = await (
+          await tables.txIdsByBlockHeight.query('', transaction.blockHeight)
+        )
+          .reverse()
+          .map(({ [RANGE_ID_KEY]: id }) => id)
+        if (!txidsByTransactionBlockHeight.includes(transaction.txid)) {
+          await tables.txIdsByBlockHeight.insert('', {
+            [RANGE_ID_KEY]: transaction.txid,
+            [RANGE_KEY]: transaction.blockHeight
+          })
+        }
+
+        // Save transaction
+        await tables.txById.insert('', transaction.txid, transaction)
+      })
     },
 
     numTransactions(): number {
-      return 0
+      return baselets.all.txsByDate.size('')
     },
 
     async removeTransaction(_txId: string): Promise<void> {
       return
     },
 
-    async updateTransaction(_args: UpdateTransactionArgs): Promise<void> {
-      return
+    async updateTransaction(args: UpdateTransactionArgs): Promise<void> {
+      const { data, txId } = args
+      return await baselets.tx(async tables => {
+        const [tx]: IProcessorTransaction[] = await tables.txById.query('', [
+          txId
+        ])
+        if (data.blockHeight != null && data.blockHeight !== tx.blockHeight) {
+          await tables.txIdsByBlockHeight.delete('', tx.blockHeight, txId)
+          await tables.txIdsByBlockHeight.insert('', {
+            [RANGE_ID_KEY]: txId,
+            [RANGE_KEY]: data.blockHeight
+          })
+
+          tx.blockHeight = data.blockHeight
+        }
+
+        await tables.txById.insert('', txId, tx)
+      })
     },
 
     async fetchTransactions(
-      _args: FetchTransactionArgs
+      args: FetchTransactionArgs
     ): Promise<IProcessorTransaction[]> {
-      return []
+      const txs: IProcessorTransaction[] = []
+      await baselets.tx(async tables => {
+        const { blockHeightMax, txId, options } = args
+        let { blockHeight } = args
+        // Fetch transactions by id
+        if (txId != null) {
+          try {
+            const txById: IProcessorTransaction[] = await tables.txById.query(
+              '',
+              [txId]
+            )
+            txs.push(...txById)
+          } catch {}
+        }
+        // Fetch transactions by min block height
+        if (blockHeightMax != null && blockHeight == null) blockHeight = 0
+        if (blockHeight != null) {
+          try {
+            const txidsByMinBlockHeight = await (
+              await tables.txIdsByBlockHeight.query(
+                '',
+                blockHeight,
+                blockHeightMax
+              )
+            )
+              .reverse()
+              .map(({ [RANGE_ID_KEY]: id }) => id)
+
+            const txsById: IProcessorTransaction[] = await tables.txById.query(
+              '',
+              txidsByMinBlockHeight
+            )
+            txs.push(...txsById)
+          } catch {}
```

Comment:
Thanks for bringing this up. This was also not properly done in the current/deprecated processor. There we sometimes check the exception message:
```
  try {
    // Create index for size of value
    await tables.utxoIdsBySize.insert('', ...)
  } catch (err) {
    if (err.message !== 'Cannot insert data because id already exists') {
      throw err
    }
```
And sometimes we just ignored errors, like error. It's also why the error handling in the tests is wonky to say the least. There should be a middle ground between not throwing and throwing only when something is really wrong, and it's not just that we don't have any results for a certain key.  I'll research some more how we could go about typing the errors..